ocaml-lsp-server/src/state.mli

```diff
@@ -13,7 +13,6 @@ type t =
   ; trace : TraceValue.t
   ; ocamlformat : Fmt.t
   ; diagnostics : Diagnostics.t
-  ; dune : Dune.t
```

Comment:
Interesting.. That's because the dune instance is not used anywhere except in the `start` function?
FWIU, ocamlformat and dune have a similar lifecycle, can't we get rid of `ocamlformat` in the state as well?

Reply:
> ocamlformat and dune have a similar lifecycle, can't we get rid of ocamlformat in the state as well?

It's possible but more difficult. We need an instance of ocamlformat to serve the formatting of requests.

I would like to get rid of this state thing eventually, but a proper design to replace it needs to come up first.